sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridPublisherClientBuilder.java

```diff
@@ -168,7 +159,7 @@ public EventGridPublisherClientBuilder() {
             .build();
 
 
-        return new EventGridPublisherAsyncClient<T>(buildPipeline, hostname, buildServiceVersion, eventClass);
+        return new EventGridPublisherAsyncClient<T>(buildPipeline, endpoint, buildServiceVersion, eventClass);
```

Comment:
Update the javadoc as well to make it clear that the endpoint should include the API endpoint as well and not just the hostname.

Reply:
Added "It must be the full url of the endpoint instead of just the hostname." in javadoc